lib/sift/where_handler.rb

```diff
@@ -18,13 +18,20 @@ def apply_jsonb_conditions(collection, value)
       return collection.where("#{@param.internal_name} @> ?", val.to_s) if value.is_a?(Array)
 
       value.each do |key, val|
-        condition = if val.is_a?(Array)
-          "('{' || TRANSLATE(#{@param.internal_name}->>'#{key}', '[]','') || '}')::int[] && ARRAY[?]"
-        else # Single Value
-          val = val.to_s
-          "#{@param.internal_name}->>'#{key}' = ?"
+        collection = if val.is_a?(Array)
+          if val.include?(nil)
+            condition = val.each_with_index.map do |element, i|
+              "#{@param.internal_name}->>'#{key}' #{element === nil ? 'IS NULL' : "= :value_#{i}"}"
+            end.join(' OR ')
```

Comment:
In this case the conditions are joined with OR. Is that going to lead to unexpected results compared to the else case below?
If case:
```
condition = val.each_with_index.map do |element, i|
  "#{@param.internal_name}->>'#{key}' #{element === nil ? 'IS NULL' : "= :value_#{i}"}"
end.join(' OR ')
```

Else case: 
`condition =  "('{' || TRANSLATE(#{@param.internal_name}->>'#{key}', '[]','') || '}')::text[] && ARRAY[?]"`

I'm not super familiar with the JSON query syntax to understand what that else condition is doing.

Reply:
The **If** condition is specifically designed for filtering the records that has `null` in the JSONB value, for example:
`metadata[verified]=[false, null]` will return:

```
|   id   |   metadata               |
-------------------------------------
|   2    |  {}                      |
|   3    |  {"verified" => false}   |
```
_(consider the record `id=1` has `{"verified" => true}`)_


The **Else** is to filter by inclusion when the JSONB value is composed by an array, example:
`metadata[ingredients]=[12,13,15]` will return:
```
|   id   |   metadata                    |
------------------------------------------
|   2    |  {"ingredients" => [12]}      |
|   3    |  {"ingredients" => [13]}      |
|   8    |  {"ingredients" => [13,20]}   |
|   9    |  {"ingredients" => [13,15]}   |
```

Reply:
so if someone queried `metadata[ingredients]=[12,13,15, null]` it would return the following?
```
|   id   |   metadata                    |
------------------------------------------
|   2    |  {"ingredients" => [12]}          |
|   3    |  {"ingredients" => [13]}          |
|   10    |  {"ingredients" => [null]}      |
```

Reply:
@amayer171 the PR has been updated, now the resultant condition will return the same results when a `nil` value is included, also it will return the records with `nil` in the "ingredients" key:
```
(
  (('{' || TRANSLATE(metadata->>'ingredients', '[]','') || '}')::text[] && ARRAY['12','13','15'])
  OR
  (
    metadata->>'ingredients' = '12'
    OR metadata->>'ingredients' = '13'
    OR metadata->>'ingredients' = '15'
    OR metadata->>'ingredients' IS NULL
  )
)
```